src/main/compa/daos/UserDAO.java

```diff
@@ -1,57 +1,72 @@
-    Logger logger = Logger.getLogger("user_dao");
+    private Logger logger = Logger.getLogger("user_dao");
 
-    public UserDAO(Datastore ds){
-        super(User.class, ds);
+    public UserDAO(Container container){
+        super(User.class, container);
     }
 
-    public User getByLoginAndPassword(String login, String password){
-        return this.createQuery().filter("login", login).
-                filter("password", password).get();
+    public void getByLoginAndPassword(String login, String password, Handler<AsyncResult<User>> resultHandler){
+        vertx.executeBlocking( future -> {
+            future.complete(this.createQuery().filter("login", login).filter("password", password).get());
+        }, resultHandler);
     }
 
-    public User addUser(String login, String password) throws RegisterException {
-        User user = this.createQuery().filter("login", login).get();
+    public void addUser(String login, String password, Handler<AsyncResult<User>> resultHandler) {
 
-        if(password.length() < PASSWORD_MIN_LENGTH)
-            throw new RegisterException(RegisterException.PASSWORD_TOO_SHORT);
+        vertx.executeBlocking( future -> {
+            User user = this.createQuery().filter("login", login).get();
 
-        if(user != null)
-            throw new RegisterException(RegisterException.USER_ALREADY_EXIST);
+            if(password.length() < PASSWORD_MIN_LENGTH)
+                future.fail(new RegisterException(RegisterException.PASSWORD_TOO_SHORT));
+
+            if(user != null)
+                future.fail(new RegisterException(RegisterException.USER_ALREADY_EXIST));
+
+            user = new User(login, password);
+            this.save(user);
+            future.complete(user);
+
+        }, resultHandler);
 
-        user = new User(login, password);
-        this.save(user);
-        return user;
     }
 
-    @Override
-    public User findOne(String key, Object value){
-        logger.log(Level.INFO, "Looking for user by key {0} and value {1}", new Object[]{key, value});
-        User u = super.findOne(key, value);
-        logger.log(Level.INFO, "User {0} found", u == null ? "not" : "");
-        return u;
+    public void findOne(String key, Object value, Handler<AsyncResult<User>> resultHandler){
+
+        vertx.executeBlocking( future -> {
+            logger.log(Level.INFO, "Looking for user by key {0} and value {1}", new Object[]{key, value});
+            User u = super.findOne(key, value);
+            logger.log(Level.INFO, "User {0} found", u == null ? "not" : "");
+            future.complete(u);
+
+        }, resultHandler);
+
     }
 
-    @Override
-    public User findById(String id) {
-        logger.log(Level.INFO, "Looking for user {0}", id);
-        User u = super.findById(id);
-        logger.log(Level.INFO, "User {0} found", u == null ? "not" : "");
-        return u;
+    public void findById(String id, Handler<AsyncResult<User>> resultHandler) {
```

Comment:
@ssssarah Is it possible to delegate this async job to the DAO ? I mean, the call of this `vertx.executeBlocking` is recurrent and it's called on each method. Shouldn't we call a generic function which will run our blocking method ?

Reply:
I like the idea, but I'd have to look into it because I don't have an immediate answer on how to implement that